### PR TITLE
fixes to react aspect docs

### PR DESCRIPTION
**Pull request by benjgil**
remove references to services as extension APIs 
remove providers - API isn't ready yet and current extensions don't support providers
fixes for other areas of explanation

### CODE REVIEW

File: docs/aspects/teambit.react/react/react.docs.md [hunk 1]
```diff
@@ -132,23 +137,21 @@ export class CustomReactExtension {
...
-> To override any specific configuration it's recommended to create a config file for the specific tool and import it to any of the **transformers**.
+> To override any specific configuration it's recommended to create a separate config file with your customized configuration and import it into the main `.extension.ts` file where you can supply it to the relevant **transformers**. See the tsconfig example in the above snippet.
 
 ### Composing tools and services
 
-Environments use Environment Services by implementing a special class of methods called Service Handlers.  
-You can override an environment's compiler replacing its Compiler Service Handler with the method `getCompiler()`.  
+Environments use Environment Services by implementing a special class of methods called Service Handlers.
+The supplied transformers allow overriding of these services - for instance the `overrideCompiler` transformer allows overriding the environment's `getCompiler()` method.
 [Find all service handlers here](#service-providers-api-docs).
```
Comment (itaymendel) [inline docs/aspects/teambit.react/react/react.docs.md]: if you commented the APIs, this link will be broken. so perhaps remove it?

File: docs/aspects/teambit.react/react/react.docs.md [hunk 2]
```diff
@@ -132,23 +137,21 @@ export class CustomReactExtension {
...
 
-The below example uses a Service Handler to change compilation service.
+The below example uses the overrideCompiler transformer to override the `getCompiler()` Service Handler to change the compilation service.
```
Comment (itaymendel) [inline docs/aspects/teambit.react/react/react.docs.md]: `overrideCompiler`

File: docs/aspects/teambit.react/react/react.docs.md (hunk 1 above)
Comment (benjgil) [reply docs/aspects/teambit.react/react/react.docs.md]: Right, removing.

File: docs/aspects/teambit.react/react/react.docs.md (hunk 2 above)
Comment (benjgil) [reply docs/aspects/teambit.react/react/react.docs.md]: nice catch (though the comment's rather subtle - worth mentioning that you're only suggesting adding overlay)

**Review by itaymendel — APPROVED** [review]

**Pull request merged**